### PR TITLE
Conditional constraint template

### PR DESCRIPTION
**Pull request by suryatejaboorlu**
The customer doesn't want to install the default constraint templates. Hence, created a conditional variable.

### CODE REVIEW

File: packages/rancher-gatekeeper/generated-changes/overlay/templates/allowedrepos.yaml
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/overlay/templates/allowedrepos.yaml:1]: nit: add space to the end of all of these Helm blocks. e.g.

```suggestion
{{- if .Values.installDefaultConstraintTemplates }}
```

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
nit review comments.

Let's also refactor these commits to the following
1. "Bump rancher-gatekeeper to 100.0.2": add package.yaml changes
2. "Allow users to disable default constraint templates": add generated-changes changes
3. "make charts": add all change to charts/, assets/ and index.yaml

Once we refactor to these commit and make the nit changes to the chart shown below, this PR should be good to go.

Will need a second review from a chart SME before merge, added @cmurphy for that review.

**Comment by nicholasSUSE** [comment]
Closed due to inactivity and lack of labels

**Pull request closed without merging**
